src/ui/server/middleware/insurance/application-access/index.test.ts

```diff
@@ -0,0 +1,98 @@
+import applicationAccessMiddleware, { IRRELEVANT_ROUTES } from '.';
+import { INSURANCE_ROUTES } from '../../../constants/routes/insurance';
+import { mockReq, mockRes, mockApplication } from '../../../test-mocks';
+import { Next, Request, Response } from '../../../../types';
+
+const { INSURANCE_ROOT, PAGE_NOT_FOUND, ELIGIBILITY, ACCOUNT, DASHBOARD, NO_ACCESS_TO_APPLICATION, ALL_SECTIONS } = INSURANCE_ROUTES;
+
+describe('middleware/insurance/application-access', () => {
+  let req: Request;
+  let res: Response;
+  let next: Next;
+
+  const nextSpy = jest.fn();
+
+  beforeEach(() => {
+    req = mockReq();
+    res = mockRes();
+    next = nextSpy;
+
+    res.locals.application = mockApplication;
+  });
+
+  describe('IRRELEVANT_ROUTES', () => {
+    it('should return an array of routes', () => {
+      const expected = [
+        PAGE_NOT_FOUND,
+        ...Object.values(ELIGIBILITY),
+        ...Object.values(ACCOUNT.CREATE),
+        ...Object.values(ACCOUNT.SIGN_IN),
+        ...Object.values(ACCOUNT.RESET_PASSWORD),
+        DASHBOARD,
+        NO_ACCESS_TO_APPLICATION,
+      ];
+
+      expect(IRRELEVANT_ROUTES).toEqual(expected);
+    });
+  });
+
+  describe('when the route is not relevant', () => {
+    beforeEach(() => {
```

Comment:
do we need the beforeEach as there is only 1 test in each describe

Reply:
We do because there's a spy for this test, the spy won't work if it's not in a `beforeEach`